File: csharp/ChainOfResponsibility-Pets/Pet.cs
public class Pet
{
    protected Pet nextPet;

    public Pet NextPet
    {
        get => nextPet;
        set => nextPet = value ?? throw new ArgumentNullException(nameof(value));
    }

    public Pet(Pet nextPet)
    {
        this.nextPet = nextPet;
    }

    public virtual string HandleTrap(Bait bait)
    {
        throw new NotImplementedException();
    }


}

File: java/src/main/java/codingdojo/Pet.java
package codingdojo;

import org.apache.commons.lang3.NotImplementedException;

public class Pet {
    protected Pet nextPet;
    public Pet(Pet nextPet) {
        this.nextPet = nextPet;
    }
    public String handleTrap(Bait bait) {
        throw new NotImplementedException("That is an unknown pet");
    }
}
